tests/storage/test_event_federation.py

```diff
@@ -259,6 +259,169 @@ def insert_event(txn):
         )
         self.assertSetEqual(difference, set())
 
+    def test_auth_difference_partial_cover(self):
+        """Test that we correctly handle rooms where not all events have a chain
+        cover calculated. This can happen due to a downgrade/upgrade.
```

Comment:
If we move these to schema 59 is the downgrade really a potential cause of this? It is likely worth keeping the code to handle this case regardless.

Reply:
Yeah, I've updated it